### PR TITLE
chore: wandb_backend_spy run config; migrate test_launch.py

### PR DESCRIPTION
**Pull request by timoffex**
Description
---
Allows inspecting the run config received by the backend in tests and migrates one test from `relay_server` to `wandb_backend_spy`.

### CODE REVIEW

File: tests/system_tests/wandb_backend_spy/spy.py [hunk 1]
```diff
@@ -34,24 +32,61 @@ def freeze(self) -> Iterator[WandbBackendSnapshot]:
...
+        if "mutation UpsertBucket" not in query:
+            return
+
+        if "config" not in variables:
+            return
+
+        run_id = variables["name"]
+        config = variables["config"]
+
+        print(f"UPDATED CONFIG {run_id}")
```
Comment (kptkin) [inline tests/system_tests/wandb_backend_spy/spy.py:70]: hmm... probably want to remove the print?

File: tests/system_tests/wandb_backend_spy/spy.py [hunk 2]
```diff
@@ -34,24 +32,61 @@ def freeze(self) -> Iterator[WandbBackendSnapshot]:
...
+        request = json.loads(request_raw)
+
+        with self._lock:
+            query: str | None = request.get("query")
+            variables: dict[str, Any] | None = request.get("variables")
+            if query is None or variables is None:
+                return
+
+            print("GOT QUERY: ", query)
+            print("WITH VARIABLES: ", variables)
```
Comment (kptkin) [inline tests/system_tests/wandb_backend_spy/spy.py:50]: remove the print? should we just have logs if we want to capture this info?

File: tests/system_tests/wandb_backend_spy/spy.py (hunk 1 above)
Comment (timoffex) [reply tests/system_tests/wandb_backend_spy/spy.py:70]: oh, oops, good catch

**Review by kptkin — APPROVED** [review]

File: tests/system_tests/wandb_backend_spy/spy.py (hunk 2 above)
Comment (timoffex) [reply tests/system_tests/wandb_backend_spy/spy.py]: removed!

Comment (jacobromero) [inline tests/system_tests/wandb_backend_spy/spy.py]: We should probably remove these 🙂

File: tests/system_tests/wandb_backend_spy/spy.py (hunk 1 above)
Comment (jacobromero) [inline tests/system_tests/wandb_backend_spy/spy.py]: another print statement here as well

File: tests/system_tests/wandb_backend_spy/spy.py (hunk 2 above)
Comment (timoffex) [reply tests/system_tests/wandb_backend_spy/spy.py]: one step ahead of you :) already removed

**Review by jacobromero — APPROVED** [review]

**Pull request merged**